library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopFederatedIT.java

```diff
@@ -0,0 +1,284 @@
+    public static final String SOFTWARE_GROUP = "software";
+    public static final String PERSON_GROUP = "person";
+    public static final String CREATED_EDGE_GROUP = "created";
+    public static final String NAME_PROPERTY = "name";
+    public static final String WEIGHT_PROPERTY = "weight";
+
+    @Before
+    public void setUp() throws Exception {
+        federatedGraph = GafferPopFederatedTestUtil.setUpFederatedGraph(GafferPopFederatedIT.class);
+        gafferPopGraph = GafferPopGraph.open(TEST_CONFIGURATION_1, federatedGraph);
+    }
+
+    @After
+    public void tearDown() {
+        CacheServiceLoader.shutdown();
+    }
+
+    @Test
+    public void shouldConstructFederatedGafferPopGraph() {
+        // When
+        final Map<String, Object> variables = gafferPopGraph.variables().asMap();
+
+        // Then
+        assertThat(variables.get(GafferPopGraphVariables.SCHEMA))
+                .isSameAs(federatedGraph.getSchema());
+
+        assertThat(variables.get(GafferPopGraphVariables.USER))
+                .hasFieldOrPropertyWithValue("userId", "user01");
+    }
+
+    @Test
+    public void shouldGetVerticesById() {
+        // Given
+        GraphTraversalSource g = gafferPopGraph.traversal();
+
+        // When
+        List<Map<Object, Object>> vertex1 = g.V(VERTEX_PERSON_1).elementMap().toList();
+        List<Map<Object, Object>> vertex2 = g.V(VERTEX_SOFTWARE_2).elementMap().toList();
+
+        // expected
+        Map<Object, String> expectedVertex1 = new HashMap<>();
+        expectedVertex1.put(T.id, VERTEX_PERSON_1);
+        expectedVertex1.put(T.label, PERSON_GROUP);
+        expectedVertex1.put("name", "person1Name");
+
+        Map<Object, String> expectedVertex2 = new HashMap<>();
+        expectedVertex1.put(T.id, VERTEX_SOFTWARE_2);
+        expectedVertex1.put(T.label, SOFTWARE_GROUP);
+        expectedVertex1.put("name", "software2Name");
+
+        // Then
+        assertThat(vertex1)
+                .hasSize(1)
+                .first()
+                .isEqualTo(expectedVertex1);
+
+        assertThat(vertex2)
+                .hasSize(1)
+                .first()
+                .isEqualTo(expectedVertex2);
+    }
+
+    @Test
+    public void shouldGetVertexPropertyValue() {
+        // Given
+        GraphTraversalSource g = gafferPopGraph.traversal();
+
+        // When
+        List<Object> vertex = g.V(VERTEX_SOFTWARE_1).values("name").toList();
+
+        // Then
+        assertThat(vertex)
+                .hasSize(1)
+                .first()
+                .hasToString("software1Name");
+    }
+
+    @Test
+    public void shouldFilterVertexesByPropertyValue() {
+        // Given
+        GraphTraversalSource g = gafferPopGraph.traversal();
+
+        // When
+        List<Object> result = g.V().outE(CREATED_EDGE_GROUP).has(WEIGHT_PROPERTY, P.gt(0.4)).values(WEIGHT_PROPERTY)
+                .toList();
+
+        // Then
+        assertThat(result)
+                .hasSize(2)
```

Comment:
You don't need a `hasSize()` assert if you use something like `containsExactlyInAnyOrder()` instead.